source/dub/commandline.d

```diff
@@ -257,6 +257,7 @@ struct CommonOptions {
 	/// Parses all common options and stores the result in the struct instance.
 	void prepare(CommandArgs args)
 	{
+		registry_urls = environment.get("DUB_REGISTRY", null).split(";");
```

Comment:
OK, makes sense to move to dub.d then. Abstracting out environment could indeed make sense, but having a library react to env vars has it's use-case as well (e.g. CURL).